api/bootstrap/kubeadm/v1beta2/kubeadm_types.go

```diff
@@ -242,6 +242,7 @@ type APIEndpoint struct {
 	// bindPort sets the secure port for the API Server to bind to.
 	// Defaults to 6443.
 	// +optional
+	// +kubebuilder:validation:Minimum=1
```

Comment:
I assume the `0` value for bind would mean random port and wouldn't be appropriate here?

Reply:
Basically if it even works APIserver would bind on a random port and then the later Nodes would try to join on a random port